### PR TITLE
Increase max connections to reduce chance of eclipse attacks

### PR DESCRIPTION
**Pull request by ghost**
If there are still a large number of non-classic nodes when the hard fork activates regular nodes may create what is effectively an eclipse attack, this should significantly reduce the likelihood of that happening.

### CODE REVIEW

File: src/net.cpp [hunk 1]
```diff
@@ -59,7 +59,7 @@
 using namespace std;
 
 namespace {
-    const int MAX_OUTBOUND_CONNECTIONS = 8;
+    const int MAX_OUTBOUND_CONNECTIONS = 32;
```
Comment (orangeman) [inline src/net.cpp:62]: How could we miss this?
Thanks for pointing it out!

**Comment by zander** [comment]
Please explain why you think this is better, and please give some testing output you gathered that led you to these magical values :)

I'm sceptical that this will work on NATed nodes, what do you think about that?

**Comment by dgenr8** [comment]
NACK.  The key is to spend less time reviewing these than it takes to create them.

**Comment by ghost** [comment]
@zander The MAX_OUTBOUND_CONNECTIONS variable increases the amount of connections being established from nodes even those behind NAT which increases the likelyhood of getting a connection with another node that will relay blocks on the new chain. The DEFAULT_MAX_PEER_CONNECTIONS change increases the amount of allowable inbound connections for those that are not behind NAT close to the select() limit so that nodes accepting connections have plenty of connection space for nodes behind NAT.

**Comment by circeng** [comment]
It would be superior to program Classic to not connect to Core useragents.

**Comment by jtoomim** [comment]
Merge window is open only for 2 MB HF stuff.

An exception for this PR is not justified. A [vote](https://bitcoinclassic.consider.it/merge-20-increase-max-connections-to-reduce-chance-of-eclipse-attacks?results=true) has been started on consider.it. It does not appear to be a favorable change. 

I am closing this. It may be reopened after the fork.

**Comment by petertodd** [comment]
Concept ACK

Improving interconnectivity during hard forks is pretty important. (but there may be more efficient ways to do it)

**Comment by jtoomim** [comment]
Come on, Peter, you should know better than that. Increasing the number of max connections totally trashes block propagation, due to the fact that blocks are uploaded to many peers in parallel.

http://rusty.ozlabs.org/?p=509

If you're going to try to get us to merge a stupid change, can you at least be more subtle about it?

**Comment by petertodd** [comment]
> Come on, Peter, you should know better than that. Increasing the number of max connections totally trashes block propagation, due to the fact that blocks are uploaded to many peers in parallel.
> 
> http://rusty.ozlabs.org/?p=509
> 
> If you're going to try to get us to merge a stupid change, can you at least be more subtle about it?

That's why I said "Concept ACK", noting that this addresses a real issue, but there may be more efficient ways to do it. If I had thought this change should be merged as-is, I'd have said ACK or utACK.

**Pull request closed without merging**